lib/streamlit/web/server/server.py

```diff
@@ -781,21 +781,11 @@ def on_message(self, payload: Union[str, bytes]) -> None:
                 )
 
             msg.ParseFromString(payload)
-            msg_type = msg.WhichOneof("type")
-
             LOGGER.debug("Received the following back message:\n%s", msg)
 
-            if msg_type == "rerun_script":
-                self._session.handle_rerun_script_request(msg.rerun_script)
-            elif msg_type == "load_git_info":
-                self._session.handle_git_information_request()
-            elif msg_type == "clear_cache":
-                self._session.handle_clear_cache_request()
-            elif msg_type == "set_run_on_save":
-                self._session.handle_set_run_on_save_request(msg.set_run_on_save)
-            elif msg_type == "stop_script":
-                self._session.handle_stop_script_request()
-            elif msg_type == "close_connection":
+            if msg.WhichOneof("type") == "close_connection":
+                # "close_connection" is a special developmentMode-only
+                # message used in e2e tests to test disabling widgets.
```

Comment:
`BackMsg.close_connection` is a special-case backmsg hack - it has nothing to do with an AppSession (unlike all other BackMsgs), and is only used in our (browser-based) e2e tests. I think it's cleaner to leave it here than to push it up to AppSession (which would require AppSession getting a Server reference).